src/Uno.Toolkit.UI/Controls/Chips/ChipGroup.cs

```diff
@@ -56,13 +56,28 @@ private void OnSelectionMemberPathChanged(DependencyPropertyChangedEventArgs e)
 			}
 		}
 
-		private void ApplyIconTemplate()
+		private void ApplyIconTemplate(DataTemplate? oldTemplate, DataTemplate? newTemplate)
 		{
-			var itemTemplate = IconTemplate;
+			if (oldTemplate == newTemplate) return;
+
 			foreach (var container in this.GetItemContainers<Chip>())
 			{
-				container.Icon = itemTemplate != null ? container.Content : null;
-				container.IconTemplate = itemTemplate;
+				if (newTemplate is not null)
+				{
+					// If there was no Icon assigned, use Content as Icon
+					// Otherwise the icon-presenter will not display anything without a icon
```

Comment:
```suggestion
					// If there was no Icon assigned, use Content as Icon
					// Otherwise the icon presenter will not display anything without an icon
```